packages/lending/src/adapter.ts

```diff
@@ -0,0 +1,991 @@
+    const protocol = this.getProtocol(protocolId);
+
+    portfolio = portfolio || (await protocol.getPortfolio(account, marketId));
+    const afterPortfolio = portfolio.clone();
+
+    // ---------- flashloan ----------
+    const flashLoanAggregatorQuotation = await apisdk.protocols.utility.getFlashLoanAggregatorQuotation(this.chainId, {
+      loans: [{ token: wrappedSrcToken, amount: srcAmount }],
+    });
+
+    const flashLoanTokenAmount = flashLoanAggregatorQuotation.loans.tokenAmountMap[wrappedSrcToken.address];
+
+    const [flashLoanLoanLogic, flashLoanRepayLogic] = apisdk.protocols.utility.newFlashLoanAggregatorLogicPair(
+      flashLoanAggregatorQuotation.protocolId,
+      flashLoanAggregatorQuotation.loans.toArray()
+    );
+    leverageShortlogics.push(flashLoanLoanLogic);
+
+    // ---------- swap ----------
+    const swapper = this.findSwapper([wrappedSrcToken, wrappedDestToken]);
+    const swapQuotation = await swapper.quote({
+      input: flashLoanTokenAmount,
+      tokenOut: wrappedDestToken,
+      slippage: defaultSlippage,
+    });
+    const swapTokenLogic = swapper.newSwapTokenLogic(swapQuotation);
+    leverageShortlogics.push(swapTokenLogic);
+
+    // ---------- supply ----------
+    const supplyLogic = await protocol.newSupplyLogic({ input: swapQuotation.output, marketId, account });
+    leverageShortlogics.push(supplyLogic);
+
+    afterPortfolio.supply(swapQuotation.output.token, swapQuotation.output.amount);
+
+    // ---------- return funds ----------
+    if (supplyLogic.fields.output) {
+      const returnLogic = apisdk.protocols.utility.newSendTokenLogic({
+        input: supplyLogic.fields.output,
+        recipient: account,
+      });
+      leverageShortlogics.push(returnLogic);
+    }
+
+    // ---------- borrow ----------
+    const borrowTokenAmount = flashLoanAggregatorQuotation.repays.tokenAmountMap[wrappedSrcToken.address];
+    const borrowLogic = protocol.newBorrowLogic({
+      output: borrowTokenAmount,
+      interestRateMode: defaultInterestRateMode,
+      marketId,
+    });
+    leverageShortlogics.push(borrowLogic);
+
+    afterPortfolio.borrow(borrowTokenAmount.token, borrowTokenAmount.amount);
+
+    // ---------- flashloan repay ----------
+    leverageShortlogics.push(flashLoanRepayLogic);
+
+    // ---------- tx related ----------
+    const estimateResult = await apisdk.estimateRouterData(
+      {
+        chainId: this.chainId,
+        account,
+        logics: leverageShortlogics,
+      },
+      this.permitType
+    );
+
+    const buildRouterTransactionRequest = (
+      args?: Omit<apisdk.RouterData, 'chainId' | 'account' | 'logics'>,
+      apiKey?: string
+    ): Promise<common.TransactionRequest> =>
+      apisdk.buildRouterTransactionRequest(
+        { ...args, chainId: this.chainId, account, logics: leverageShortlogics },
+        apiKey ? { 'x-api-key': apiKey } : undefined
+      );
+
+    return {
+      fields: {
+        srcToken,
+        srcAmount,
```

Comment:
Do we need to return destAmount too?

Reply:
After discussion will add the dest token & amount in short & long